dmutils/filters.py

```diff
@@ -24,18 +25,19 @@ def format_links(text):
                                 \.                        # match dot in host name
                                 (?:[^\s/?\.#<>"']+\.?)+   # match rest of domain
                                 (?:/[^\s<>"']*)?          # rest of url
-                                [^\s<>"'\.]               # no dot at end
+                                [^\s<>,"'\.]              # no dot at end
                                 )""", re.X)
-    matches = url_match.findall(text)
-    link = '{0}<a href={1} rel="external">{1}</a>'
-    if matches:
-        textArray = url_match.split(text)
+    matched_urls = url_match.findall(text)
+    if matched_urls:
+        link = '{0}<a href="{1}" rel="external">{2}</a>'
         formattedText = ""
-        for text in textArray:
-            if text in matches:
-                formattedText = link.format(formattedText, Markup.escape(text))
+        textArray = url_match.split(text)
+        for partial_text in textArray:
+            if partial_text in matched_urls:
+                url = "http://{}".format(partial_text) if partial_text.startswith('www') else partial_text
```

Comment:
I know what you mean about http-only, but seemed like the only way to deal with `www` links.  I'm not sure if including `www` for linking was based on an existing need for things that are already breaking our pages.  @carolinegreen knows more than me about the original issues we've seen.